### DISCUSSION TITLE
Android emulator support for Apple Silicon

### DISCUSSION THREAD

**Discussion by Reedyuk** [codemagic-ci-cd/codemagic-docs#1809] [Ideas]
### Use case

```Text
I would like to get the benefit of Apple Silicon's high performance on the ci and be able to use Android emulator.
```

### Idea

```Text
I have some integration tests that require the Android Emulator, this is currently not supports by the mac_mini_m1 + m2's
```

**Comment by Ycoi** [comment]
Thanks for the feedback. There has been issues with Android Emulator on Apple Silicon because of lack of proper support by Android emulator itself. I’ll discuss this with the team and see if we can test this with the latest versions to see if they would be reliable enough by now.

**Reply by Ycoi** [reply]
I checked this with our engineering team and unfortunately the reason why we can’t run Android emulators on our Apple Silicon  machines is that Apple virtualizarion framework doesn’t support nested virtualisation.  And before Apple adds this support, we unfortunately can’t add support for Android emulators either. 

M2 has hardware support for nested virtualization, but unfortunately macOS and Apple virtualization network still doesn’t support it. Hopefully this would chage in the future.

**Reply by Reedyuk** [reply]
thanks for the quick response. Is it worth leaving this request open until its supported?

**Reply by Ycoi** [reply]
Yes, let’s leave it open at least for now. That way we can track the request better. We can always close it later if it seems that Apple doesn’t add the support any time soon.

**Reply by Ycoi** [reply]
I thought about this a bit more and I think it makes sense to close this feature request since it doesn’t depend on us but on Apple. This is anyway so core functionality that if it just becomes possible, we will add the support.
